README.md

```diff
@@ -13,3 +13,31 @@ installed.
 ## API Documentation
 API Docs can be found at `/api/documentation` on the back-end server. Also check out the code for more explanations
 and documentation! We've tried to keep it up-to-date.
+
+## Runbook
+This section is to collect thoughts/learnings from the codebase that have been hard-won, so we don't lose a record of it
+if and when the information proves useful again
+
+### Derived fields
+Normally, derived fields on models are represented as `@property`-decorated functions. However, there are a few in
+the codebase that need to be accessed on the database layer without `JOIN`s. So that they can be indexed.
+Specifically, these are the `full_code` fields on `Course` and `Section` models, which are derived from fields on related
+models.
+
+These are updated every time the `save()` method is called. However, it's possible to get into a state 
+(such as with db migrations) where `full_code` isn't set properly.
+
+Open a shell in production, and run this small script:
+```python
+from tqdm import tqdm
+from courses.models import Section, Course
+
+for c in tqdm(Course.objects.all().select_related("department")):
+    c.save()
+
+for s in tqdm(Section.objects.all().select_related("course")):
+    s.save()
+```
```

Comment:
do you need to activate a virtual env to run this in prod or nah, not familiar with how y'all got everything set up tbh

Reply:
Nope! No virtualenv in prod. @pawalt probably should write a user guide for Bastion (the system for sshing into prod), but I know that it's going thru a lot of changes this semester

Reply:
This is dockerfile-dependent. If you have the `--system` flag on your pipenv install, then you don't need a venv, otherwise you have to do a `pipenv run ./manage.py`. Bastion is changing but this will not change.